src/utils/utils-contour.js

```diff
@@ -91,35 +105,52 @@ export const getContourMarks = (layerName, state) => [
         field: "y"
       },
       strokeColor: {
-        scale: "contour_color",
-        field: "is_major"
+        scale: CONTOUR_COLOR_SCALE,
+        field: state.encoding.color.field
```

Comment:
so now that we've moved to defining the `field` explicitly in the vega lite json, we need to verify that "field" exists, otherwise we're prone to setting field incorrectly. If field doesn't exist in the color encoding, then it seems this would fill out to "undefined" (or crash if state.encoding.color doesn't exist), resulting in a backend render error. Similarly with stroke width below.

It seems that as of now we can assume that the field here for both must be the `isMajorFieldName`. Perhaps we can use that as a default for now too? Seems a little backwards to use that as a default, but it makes sense at the same time. Anyway, seems like we need some protection here. Otherwise looks good.

NOTE: this kind of stuff is why I had started developing a render-vega-lite module in the first place that would automatically handle logic for this, i.e. including the scales and validating them when requested rather than the dev having to account for it in their own individual layer.

Reply:
Is that all the stuff you added with wind barbs? Maybe I can try out using it in the follow up ticket

Reply:
yes, and it is going to be exposed through a more general utility with an upcoming PR (https://github.com/heavyai/heavyai-charting/tree/chris/render_vega_lite) with the goal of it probably turning into it's own repo. But I wouldn't advise trying to migrate over to it on your own quite yet unless you want to experiment. It is going to be a bit tricky, and delicate, because of so much existing code bleed with the previous format (as evidenced by the whole stacked-legend fiasco). I'm happy to give you some background on it when the dust settles if you want to assist with the migration.

Reply:
Nice! I'd be happy to be a guinea pig at some point